main/src/library/File.flix

```diff
@@ -364,4 +364,152 @@ namespace File {
             }
         }
 
+    ///
+    /// Writes `data` to the given file `f`.
+    ///
+    /// Creates file `f` if it does not exist. Overwrites it if it exists.
+    ///
+    /// Returns `true` if the file `f` was created, and `false` if `f` was overwritten.
+    ///
+    pub def write(f: String, data: t): Result[Bool, String] & Impure with ToString[t] =
+        try {
+            import new java.io.File(String) as newFile;
+            import new java.io.PrintWriter(##java.io.File) as newPrintWriter;
+            import java.io.PrintWriter.println(String);
+            import java.io.PrintWriter.checkError();
+
+            let alreadyExists = File.exists(f);
+
+            let javaFile = newFile(f);
+            let w = newPrintWriter(javaFile);
+
+            println(w, "${data}");
+
+            if (checkError(w)) {
+                Err("An error happened when writing to the file: "+f)
```

Comment:
Does it make sense to checkError instead of simply closing the file using [PrintWriter.close](https://docs.oracle.com/javase/7/docs/api/java/io/PrintWriter.html#close()) ?
And if yes, should there be a better error message?

Reply:
I think we need to do both.

I don't think we can do much better than the message you provided, since `checkError` doesn't give any useful information.

Maybe "occurred" instead of "happened" though. A little more formal :)